### PR TITLE
STEP-1207: Modified 'Should retry tests on failure?' test input

### PR DESCRIPTION
**Pull request by Bence1001**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MAJOR_ [version update](https://semver.org/).

### Context
**We are adding Xcode 13's Test Repetition features to the Xcode Test step.**
* This includes choosing test repetition mode, setting maximum test repetitions and rerunning tests in a different process.
* We are removing the previously added Test Repetition behavior from the `should_retry_test_on_fail` input.

### Changes
**Modified the `should_retry_test_on_fail` step input.**
* Changed title to `Should retry tests on failure? (Not available in Xcode 13+)`.
* Changed description according to behavior changes (see below).
* No longer uses 'Retry on Failure' Test Repetition Mode if running on Xcode 13+. This has already been removed in this PR: https://github.com/bitrise-steplib/steps-xcode-test/pull/173.

### CODE REVIEW

File: xcodebuild.go [hunk 1]
```diff
@@ -283,21 +283,21 @@ func handleTestRunError(prevRunParams testRunParams, prevRunResult testRunResult
...
 				return cleanOutputDirAndRerunTest(prevRunParams)
 			}
 
-			log.Errorf("retryOnTestRunnerError=false, no more retry, stopping the test!")
+			log.Errorf("Automatic retry is disabled, no more retry, stopping the test!")
 			return prevRunResult.xcodebuildLog, prevRunResult.exitCode, prevRunResult.err
 		}
 	}
 
-	if prevRunParams.xcodeMajorVersion < 13 && prevRunParams.buildTestParams.RetryTestsOnFailure {
```
Comment (Bence1001) [inline xcodebuild.go:298]: The `prevRunParams.xcodeMajorVersion < 13` check is no longer required as it's already performed in `ProcessConfig()`. By the time this line is executed, the check should have already been performed.

File: step.yml [hunk 2]
```diff
@@ -186,12 +187,13 @@ inputs:
...
       description: |-
         The maximum number of times a test will repeat based on Test Repetition Mode.
-        
+
         Should be more than 1 if the Test Repetition Mode (`test_repetition_mode`) is other than `none`.
       is_required: true
   - relaunch_tests_for_each_repetition: "no"
     opts:
       title: Relaunch Tests for Each Repetition (Available in Xcode 13+)
+      category: Test Repetition
```
Comment (Bence1001) [inline step.yml:196]: Forgot to add `relaunch_tests_for_each_repetition` to the `Test Repetition` category in the previous PR.

**Review by ofalvai — DISMISSED** [review]
LGTM 👌

File: step.yml [hunk 3]
```diff
@@ -298,10 +300,12 @@ inputs:
   - should_retry_test_on_fail: "no"
     opts:
       category: Debug
-      title: "Should retry test on failure?"
+      title: "Should retry tests on failure? (Not available in Xcode 13+)"
+      summary: If you set this input to `yes`, the Step will rerun your tests in case of failure.
```
Comment (BanyikAnna) [inline step.yml:304]: how about this: " If you set this input to 'yes', the Step will rerun your failed tests."

File: step.yml [hunk 4]
```diff
@@ -298,10 +300,12 @@ inputs:
...
       description: |-
-       If you set this input to `yes`, the Step will rerun your tests. With Xcode 13 and above, only your failed test cases will be rerun. With older Xcode versions, all test cases will be rerun.
-        Please note that this feature is only available from Xcode 13 and above, earlier versions support rerunning ALL test cases instead of a specific failed one.
+        If you set this input to `yes`, the Step will rerun your tests once in case of failure. Note that this will rerun ALL of your tests and just the ones that failed.
```
Comment (BanyikAnna) [inline step.yml:306]: how about this: "If you set this input to 'yes', the Step will rerun ALL your tests once in the case of failed test/s. Note that ALL your tests will be rerun, not just the ones that failed."

**Review by ofalvai — APPROVED** [review]
